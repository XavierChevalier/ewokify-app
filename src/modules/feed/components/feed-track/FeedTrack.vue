<script setup lang="ts">
  import { z } from 'zod'
  import { PropType } from 'vue'
  import LikeIcon from '@/components/like-icon/LikeIcon.vue'

  defineProps({
    cover: {
      type: String,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: (value) => z.string().url().safeParse(value).success,
    },
    title: {
      type: String,
      required: true,
    },
    bands: {
      type: Array as PropType<string[]>,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: (value) => z.array(z.string()).safeParse(value).success,
    },
    duration: {
      type: Number,
      required: true,
      // eslint-disable-next-line vue/valid-define-props
      validator: (value) => z.number().gte(0).safeParse(value).success,
    },
  })
</script>

<template>
  <img :src="cover" alt="The track cover" />
  <span>{{ title }}</span>
  <span v-for="band in bands" :key="band">{{ band }}</span>
  <LikeIcon />
  <span></span>
</template>
