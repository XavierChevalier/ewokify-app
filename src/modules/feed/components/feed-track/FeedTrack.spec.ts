import FeedTrack from './FeedTrack.vue'
import fc from 'fast-check'
import { generateStorybookSnapshotTests } from '@/tests/tools/StorybookSnapshot'
import { Default } from '@/modules/feed/components/feed-track/FeedTrack.stories'
import {
  fcNotPositiveNumber,
  fcNotString,
  fcNotUrl,
  fcPositiveNumber,
} from '@/tests/tools/FastCheckProperties'
import { generatePropertyValidationTests } from '@/tests/tools/PropsValidator'

describe('FeedTrack', () => {
  generateStorybookSnapshotTests({ Default })

  generatePropertyValidationTests(FeedTrack)
    .forProperty('cover')
    .states({
      valid: fc.webUrl(),
      invalid: fcNotUrl,
      invalidSeeds: [{ seed: -355476508, path: '9:3:2:4:5:13' }],
    })
    .forProperty('bands')
    .states({
      valid: fc.array(fc.string()),
      invalid: fc.array(fcNotString, { minLength: 1 }),
    })
    .forProperty('durationMs')
    .states({
      valid: fcPositiveNumber,
      invalid: fcNotPositiveNumber,
    })
})
