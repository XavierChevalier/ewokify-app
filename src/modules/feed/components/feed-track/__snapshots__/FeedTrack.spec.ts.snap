// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedTrack [snapshot] should render the "Default" component correctly 1`] = `
<img
  alt="The track cover"
  src="https://picsum.photos/50"
/>
`;

exports[`FeedTrack should render the "Default" component correctly 1`] = `<div />`;
